docs/markdown/Java and Scala/kotlin.md

```diff
@@ -0,0 +1,223 @@
+---
+title: "Kotlin"
+slug: "kotlin"
+excerpt: "Kotlin support for Pants."
+hidden: false
+---
+
```

Comment:
This page is very, very similar to the existing page, which is a good sign!

But... rather than duplicating it, it would probably be good to extract the shared portion into a JVM overview page, and then have small pages for Kotlin, Scala, Java that highlight only the differences?

Reply:
While I would support such a refactor of the docs, I don't really have the time to devote to such a docs refactor. So I'll start with a Kotlin page in this PR so at least we have a place to link to in a Kotlin blog post with the 2.13 release.

Reply:
(Also, a follow-up to this PR will add docs on `kotlinc` plugins.)

Reply:
I continue to lean towards the best approach being something like Sphinx that lets us reuse the same docs snippet in multiple places. 

Duplication can be a good thing in docs -- users don't have to see things that aren't relevant to them, and we make our advice more localized. But it's a maintenance burden, so -> Sphinx

Reply:
These docs are markdown though, right? Where are Sphinx docs in the repo?

Reply:
We are not using sphinx yet, but it is planned as future work

Reply:
Generally speaking, I'd prefer the docs to be repetitive, so that you can scan through one page to get an idea of how things work.  The high granularity of e.g. our onboarding docs was noted as a barrier to entry at PyCon sprints.